src/main/java/org/openpnp/spi/base/AbstractPnpJobProcessor.java

```diff
@@ -84,8 +113,9 @@ public static NozzleTip findNozzleTip(Head head, Part part) throws Exception {
      * @throws Exception If no Feeder is found that is both enabled and is serving the Part.
      */
     public static Feeder findFeeder(Machine machine, Part part) throws Exception {
+		if(part!=null)
         for (Feeder feeder : machine.getFeeders()) {
-            if (feeder.getPart() == part && feeder.isEnabled()) {
+            if (feeder.getPart().getId().equals(part.getId()) && feeder.isEnabled()) {
```

Comment:
Sure that this exist. Example are parts with different height. 
Different batches have 0.3mm height differences as allowed from datasheet,
i never seen more then 0.2mm. As example Murata, if part comes from 
Germany or Hungary the part thickness is a lot different, millimeters, not thents of millimeters.
For resistor arrays, concave and convex as example have different footprint, or if part are
from differnt manufacture, description holds order part number, .... , parts are always different.